datameta/static/js/submit.js

```diff
@@ -524,18 +506,135 @@ DataMeta.submit.renderMetadataHelp = function() {
     });
 }
 
+DataMeta.submit.checkBoxes = function(table, checked) {
+    DataMeta.submit.setLock(true);
+    var checkboxes = document.querySelectorAll("input.datameta-entity-check[data-datameta-type='"+ table +"']")
+    checkboxes.forEach(function(box) {
+        box.checked = checked;
+        DataMeta.submit.unchecked[box.getAttribute('data-datameta-uuid')] = ! checked;
+    })
+
+    DataMeta.submit.submit(true);
+    DataMeta.submit.setLock(false);
+}
+
+DataMeta.submit.deleteSelectedMeta = function() {
+
+    DataMeta.submit.setLock(true);
+
+    // Get checked metadatasets
+    var metadatasetIds = Array.from(document.querySelectorAll("input.datameta-entity-check[data-datameta-type='meta']:checked"), input => input.getAttribute("data-datameta-uuid"));
+
+    // Delete all checked metadatasets
+    fetch(DataMeta.api('rpc/delete-metadatasets'),
+    {
+        method: 'POST',
+        headers: {
+            'Accept': 'application/json',
+            'Content-Type': 'application/json'
+        },
+        body:  JSON.stringify({metadatasetIds})
+    }).then(function(response) {
+        if (response.ok) {
+            return DataMeta.submit.refresh();
+        } else if (response.status==401) {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Unauthenticated.", "danger")
+        } else if (response.status==403) {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Access denied.", "danger")
+        } else if (response.status==404) {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Record not found.", "danger")
+        } else if (response.status==400) {
+            response.json().then(json => {
+                DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: " + json[0].message, "danger")
+            });
+        } else {
+            DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Unknown error.", "danger")
+        }
+        DataMeta.submit.refresh();
+    }).catch(function(error) {
+        DataMeta.new_alert("<strong>ERROR</strong> Deleting the records failed: Unknown error.", "danger")
+        console.log(error)
+    });
+
+    DataMeta.submit.setLock(false);
```

Comment:
This has to go after lines 553 and 556, otherwise the UI will be unlocked immediately before the async actions have completed.